ptychonn/pospred/configs.py

```diff
@@ -0,0 +1,290 @@
+import logging
+logging.getLogger(__name__).setLevel(logging.INFO)
+
+import collections
+import json
+import dataclasses
+import warnings
+from typing import Any
+try:
+    import tomli
+except:
+    warnings.warn('Unable to import tomli, which is needed to load a TOML config file.')
+
+@dataclasses.dataclass
+
+class ConfigDict(collections.defaultdict):
+    def __str__(self, *args, **kwargs):
+        s = ''
+        for key in self.__dict__.keys():
+            s += '{}: {}\n'.format(key, self.__dict__[key])
+        return s
+
+    def __repr__(self):
+        return self.__str__()
```

Comment:
You get `__repr__` (and `__str__`) for free when you use dataclass.

Reply:
I overrode these methods to add a linebreak after each entry to make it look prettier.